### ISSUE TITLE
relay server runs out of file descriptors

### ISSUE THREAD

**Issue by schmir-at-bb** [trustlines-protocol/relay#381]
Daniel did some light testing and the relay ran out of file descriptors:

```
Jan 20 07:36:47 trustlines-devel bash[18095]: Traceback (most recent call last):
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/gevent/threadpool.py", line 281, in _worker
Jan 20 07:36:47 trustlines-devel bash[18095]: OSError: [Errno 24] Too many open files
Jan 20 07:36:47 trustlines-devel bash[18095]: 2020-01-20T07:36:47Z (<ThreadPool at 0x7ff803b059e8 0/10/10 hub=<Hub at 0x7ff808a51c18 thread_ident=0x140703467874112>>, <built-in function getaddrinfo>) failed with OSError
Jan 20 07:36:47 trustlines-devel bash[18095]: Traceback (most recent call last):
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/gevent/threadpool.py", line 281, in _worker
Jan 20 07:36:47 trustlines-devel bash[18095]: OSError: [Errno 24] Too many open files
Jan 20 07:36:47 trustlines-devel bash[18095]: 2020-01-20T07:36:47Z (<ThreadPool at 0x7ff803b059e8 0/10/10 hub=<Hub at 0x7ff808a51c18 thread_ident=0x140703467874112>>, <built-in function getaddrinfo>) failed with OSError
Jan 20 07:36:47 trustlines-devel bash[18095]: Traceback (most recent call last):
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/gevent/threadpool.py", line 281, in _worker
Jan 20 07:36:47 trustlines-devel bash[18095]: OSError: [Errno 24] Too many open files
Jan 20 07:36:47 trustlines-devel bash[18095]: 2020-01-20T07:36:47Z (<ThreadPool at 0x7ff803b059e8 0/10/10 hub=<Hub at 0x7ff808a51c18 thread_ident=0x140703467874112>>, <built-in function getaddrinfo>) failed with OSError
Jan 20 07:36:47 trustlines-devel bash[18095]: Traceback (most recent call last):
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/gevent/baseserver.py", line 199, in _do_read
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/gevent/server.py", line 164, in do_read
Jan 20 07:36:47 trustlines-devel bash[18095]: OSError: [Errno 24] Too many open files
Jan 20 07:36:47 trustlines-devel bash[18095]: 2020-01-20T07:36:47Z <WSGIServer at 0x7ff802591550 fileno=11 address=('::', 5000, 0, 0)> failed with OSError
Jan 20 07:36:47 trustlines-devel bash[18095]: 2020-01-20 07:36:47,968[ERROR] relay.api.app: Exception on /api/v1/users/0x20BB466944583A522415e07a69C883a02B8a766E/events [GET]
Jan 20 07:36:47 trustlines-devel bash[18095]: Traceback (most recent call last):
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/flask_sockets.py", line 40, in __call__
Jan 20 07:36:47 trustlines-devel bash[18095]:     handler, values = adapter.match()
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/werkzeug/routing.py", line 1799, in match
Jan 20 07:36:47 trustlines-devel bash[18095]:     raise NotFound()
Jan 20 07:36:47 trustlines-devel bash[18095]: werkzeug.exceptions.NotFound: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.
Jan 20 07:36:47 trustlines-devel bash[18095]: During handling of the above exception, another exception occurred:
Jan 20 07:36:47 trustlines-devel bash[18095]: Traceback (most recent call last):
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
Jan 20 07:36:47 trustlines-devel bash[18095]:     rv = self.dispatch_request()
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
Jan 20 07:36:47 trustlines-devel bash[18095]:     return self.view_functions[rule.endpoint](**req.view_args)
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/flask_restful/__init__.py", line 480, in wrapper
Jan 20 07:36:47 trustlines-devel bash[18095]:     resp = resource(*args, **kwargs)
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/flask/views.py", line 89, in view
Jan 20 07:36:47 trustlines-devel bash[18095]:     return self.dispatch_request(*args, **kwargs)
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/flask_restful/__init__.py", line 595, in dispatch_request
Jan 20 07:36:47 trustlines-devel bash[18095]:     resp = meth(*args, **kwargs)
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/webargs/core.py", line 472, in wrapper
Jan 20 07:36:47 trustlines-devel bash[18095]:     return func(*new_args, **kwargs)
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/relay/src/relay/api/resources.py", line 70, in dump_result
Jan 20 07:36:47 trustlines-devel bash[18095]:     return schema.dump(wrapped(*args, **kwargs))
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/relay/src/relay/api/resources.py", line 273, in get
Jan 20 07:36:47 trustlines-devel bash[18095]:     timeout=self.trustlines.event_query_timeout,
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/relay/src/relay/relay.py", line 418, in get_user_events
Jan 20 07:36:47 trustlines-devel bash[18095]:     user_address, type, from_block
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/relay/src/relay/relay.py", line 439, in _get_network_event_queries
Jan 20 07:36:47 trustlines-devel bash[18095]:     network_address
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/relay/src/relay/relay.py", line 120, in get_event_selector_for_currency_network
Jan 20 07:36:47 trustlines-devel bash[18095]:     ethindex_db.connect(""),
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/relay/src/relay/ethindex_db.py", line 28, in connect
Jan 20 07:36:47 trustlines-devel bash[18095]:     return psycopg2.connect(dsn, cursor_factory=psycopg2.extras.RealDictCursor)
Jan 20 07:36:47 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/psycopg2/__init__.py", line 126, in connect
Jan 20 07:36:47 trustlines-devel bash[18095]:     conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
Jan 20 07:36:47 trustlines-devel bash[18095]: psycopg2.OperationalError: could not translate host name "localhost" to address: System error
Jan 20 07:36:47 trustlines-devel bash[18095]: 2020-01-20 07:36:47,980[INFO] geventwebsocket.handler: ::ffff:217.91.252.61 - - [2020-01-20 07:36:47] "GET /api/v1/users/0x20BB466944583A522415e07a69C883a02B8a766E/events HTTP/1.1" 500 196 1.309219
Jan 20 07:36:49 trustlines-devel bash[18095]: Traceback (most recent call last):
Jan 20 07:36:49 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/gevent/baseserver.py", line 199, in _do_read
Jan 20 07:36:49 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/gevent/server.py", line 164, in do_read
Jan 20 07:36:49 trustlines-devel bash[18095]: OSError: [Errno 24] Too many open files
Jan 20 07:36:49 trustlines-devel bash[18095]: 2020-01-20T07:36:49Z <WSGIServer at 0x7ff802591550 fileno=11 address=('::', 5000, 0, 0)> failed with OSError
Jan 20 07:36:49 trustlines-devel bash[18095]: Traceback (most recent call last):
Jan 20 07:36:49 trustlines-devel bash[18095]:   File "/home/devel/venv/lib/python3.6/site-packages/gevent/threadpool.py", line 281, in _worker
```

**Comment by schmir-at-bb** [comment]
I wasn't able to reproduce that issue and I'm thinking about how to proceed with this issue.

**Comment by weilbith** [comment]
Just to complete the information, I add [the used script](https://github.com/trustlines-network/mobileapp-scripts/blob/master/src/generateTestAccounts.js) here.

**Comment by berndbohmeier** [comment]
I would hope this was fixed by 3638b45fee45485fd25445cdc35e9e594eba5c9d which improved the affected endpoint. We can reopen if it occurs again.
